crates/evm/src/fuzz/mod.rs

```diff
@@ -216,6 +219,61 @@ impl<'a> FuzzedExecutor<'a> {
 
         result
     }
+
+    pub fn single_fuzz(
+        &self,
+        state: &EvmFuzzState,
+        address: Address,
+        should_fail: bool,
+        calldata: ethers::types::Bytes,
+    ) -> Result<FuzzOutcome, TestCaseError> {
+        let call = self
+            .executor
+            .call_raw(self.sender, address, calldata.0.clone(), 0.into())
+            .map_err(|_| TestCaseError::fail(FuzzError::FailedContractCall))?;
+        let state_changeset = call
+            .state_changeset
+            .as_ref()
+            .ok_or_else(|| TestCaseError::fail(FuzzError::EmptyChangeset))?;
+
+        // Build fuzzer state
+        collect_state_from_call(
+            &call.logs,
+            state_changeset,
+            state.clone(),
+            &self.config.dictionary,
+        );
+
+        // When assume cheat code is triggered return a special string "FOUNDRY::ASSUME"
+        if call.result.as_ref() == ASSUME_MAGIC_RETURN_CODE {
+            return Err(TestCaseError::reject(FuzzError::AssumeReject))
+        }
+
+        let breakpoints =
+            call.cheatcodes.clone().map_or(Default::default(), |cheats| cheats.breakpoints);
```

Comment:
We don't need to clone the entire cheatcode state for just the breakpoints

```suggestion
            call.cheatcodes.as_ref().map_or_else(Default::default, |cheats| cheats.breakpoints.clone());
```

Reply:
Or otherwise it can be `cheatcodes.take`, without cloning at all